modules/geo-layers/src/tile-layer/utils/viewport-util.js

```diff
@@ -33,10 +33,11 @@ export function getTileIndices(viewport, maxZoom, minZoom) {
     return [];
   }
 
-  viewport = new viewport.constructor({
-    ...viewport,
-    zoom: z
-  });
+  viewport = new viewport.constructor(
```

Comment:
Is the `view.constructor` part expected? (instead of new Viewport({...))

Reply:
The user-supplied viewport class is used so that this layer works with any custom (non-web-mercator) projection.